modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/AddFDSView.tsx

```diff
@@ -0,0 +1,134 @@
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import ClayAutocomplete from '@clayui/autocomplete';
+import ClayButton from '@clayui/button';
+import ClayForm, {ClayInput} from '@clayui/form';
+import ClayIcon from '@clayui/icon';
+import ClayLayout from '@clayui/layout';
+import {navigate} from 'frontend-js-web';
+import React from 'react';
+
+import '../css/AddFDSView.scss';
+
+const RequiredMark = () => (
+	<>
+		<span className="inline-item-after reference-mark text-warning">
+			<ClayIcon symbol="asterisk" />
+		</span>
+		<span className="hide-accessible sr-only">
+			{Liferay.Language.get('required')}
+		</span>
+	</>
+);
+
+type HeadlessResource = {
+	bundleLabel: string;
+	entityClassName: string;
+	name: string;
+	version: string;
+};
+interface IFDSViewsProps {
+	fdsViewsURL: string;
+	headlessResources: Array<HeadlessResource>;
+	namespace: string;
+}
+
+const AddFDSView = ({
+	fdsViewsURL,
+	headlessResources,
+	namespace,
+}: IFDSViewsProps) => {
+	return (
+		<ClayLayout.ContainerFluid formSize="lg">
+			<ClayForm>
+				<ClayLayout.Sheet className="pt-4" size="lg">
+					<ClayLayout.SheetHeader>
+						<h2 className="sheet-title">
+							{Liferay.Language.get('new-dataset')}
+						</h2>
+					</ClayLayout.SheetHeader>
+
+					<ClayLayout.SheetSection>
+						<ClayForm.Group>
+							<label htmlFor={`${namespace}fdsViewNameInput`}>
+								{Liferay.Language.get('name')}
+
+								<RequiredMark />
+							</label>
+
+							<ClayInput
+								id={`${namespace}fdsViewNameInput`}
+								type="text"
+							/>
+						</ClayForm.Group>
+
+						<ClayForm.Group>
+							<label
+								htmlFor={`${namespace}fdsViewProviderAutocomplete`}
+								id={`${namespace}fdsViewProviderAutocompleteLabel`}
+							>
+								{Liferay.Language.get('provider')}
+
+								<RequiredMark />
+							</label>
+
+							<ClayAutocomplete
+								aria-labelledby={`${namespace}fdsViewProviderAutocompleteLabel`}
+								id={`${namespace}fdsViewProviderAutocomplete`}
+								menuTrigger="focus"
+								messages={{
+									loading: '',
+									notFound: Liferay.Language.get(
+										'no-results-found'
+									),
+								}}
+								placeholder={Liferay.Language.get(
+									'choose-an-option'
+								)}
+							>
+								{headlessResources.map((headlessResource) => (
+									<ClayAutocomplete.Item
+										key={headlessResource.entityClassName}
+									>
+										{`${headlessResource.name} (${headlessResource.version} - ${headlessResource.bundleLabel})`}
+									</ClayAutocomplete.Item>
+								))}
+							</ClayAutocomplete>
```

Comment:
Hey @markocikos in this scenario you can use dynamic collection which implements list virtualization, you won't have problems when you have a big list with many items https://clayui.com/docs/components/autocomplete.html#dinamic.

If you have any pagination implementation with a filter in the backend you can also use it with the `useResource` hook. Both together offer paging and infinite scroll OOTB https://clayui.com/docs/components/autocomplete.html#asynchronous-loading. `useResource` also offers a caching system with revalidation and also the implementation of jitter in case of request failure.

Reply:
We don't have a pagination implementation with a filter in backend. I was hoping to avoid it:
- to have better UX. Async requests may take a long time, frontend only is snappier. 
- to have less code to maintain.

We may still need to implement it, if frontend-only solution ends up performing badly.

I tried the dynamic setup, `defaultItems` and `{(item) => (...)`. When comparing performance with the static solution in current PR, performance seems about the same, a minor lag first 1-2 letters. It's just my personal impression, I didn't do a precise test.

Reply:
Yeah, I agree that having the list in the frontend and doing the filter in the frontend is faster, but if that tends to grow, that number must certainly be considering data pagination, it's faster to send small pieces of the list and load on demand in the case with the infinite list than loading the whole list. 

In terms of UX, you're better off providing a quick response and loading on demand than slow loading the first time. I think this is not a bad UX, infinite list does not break the UX, the user many times will not realize that he is loading more data because the infinite list implementation many times starts requesting the next request before the scroll reaches the end of the list. But if this is a controlled and small list that is below 500 it wouldn't have a problem. Many of the APIs implement below 100 because it depends on the payload of each item.

It is worth remembering that when React has more than 100 nodes rendered in the list, it starts to have problems, it starts to be more noticeable when it reaches 100, a virtualized list will not have problems because it only renders 5 to 10 items. I would recommend using `defaultItems` instead of iterating over the list here because under the hood we iterate over the list as well to cache and pre-initialize the list. This way you will cause a quadratic iteration.

Reply:
Good to know this! I think we can push for now as it is, we'll need to improve it in a followup, in more ways than one. There is some uncertainty about potential number of items, and also we may use a different component to be more consistent with DXP solutions.